packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -302,7 +173,7 @@ export function getValidatorApi({
 
     if (skipHeadChecksAndUpdate !== true) {
       notWhileSyncing();
-      await waitForSlot(slot); // Must never request for a future slot > currentSlot
+      await waitForSlotWithDisparity(slot); // Must never request for a future slot > currentSlot
```

Comment:
what's the implication of this change?

Reply:
There is no implication apart from making it more readable and testable.